src/Illuminate/Database/Query/Builder.php

```diff
@@ -3848,7 +3848,11 @@ public function update(array $values)
 
         $values = collect($values)->map(function ($value) {
             if (! $value instanceof Builder) {
-                return ['value' => $value, 'bindings' => $value];
+                return ['value' => $value, 'bindings' => match (true) {
+                    $value instanceof Collection => $value->toArray(),
```

Comment:
Why not use `$value->all()` and skip the implicit `array_map` call inside to `Collection@toArray()`?

Reply:
Also, I'd change the check to `$value instanceof Enumerable`, so `$value->all()` also covers `LazyCollection` instances

Reply:
It might not be even worth it to pursue these changes. Using Collection as value is not in the docs (unless someone can share the URL) and Postgres/SQLServer would require different syntax to store JSON

Reply:
You're right. And it is easy enough for someone to just add `collect(...)->all()` on the call-site

Reply:
I think it's problematic though that the behaviour is somewhat undefined when passing in a collection (or at least very unexpected), especially as I think many devs don't actively differentiate between arrays and collections in Laravel as much Laravel functionality behaves the same no matter what you pass in.